### PR TITLE
generate names from swagger

### LINKED ISSUE #20707: Enum generation must be done using swagger and not generated code

### PR DESCRIPTION
**Pull request by rakshith91**
Fixes #20707

### CODE REVIEW

File: sdk/eventgrid/azure-eventgrid/swagger/postprocess_eventnames.py [hunk 1]
```diff
@@ -1,44 +1,42 @@
...
-            event_name = re.findall("Microsoft.[a-zA-Z]+.[a-zA-Z]+", event[1].__doc__)[0]
-        except:
-            # these two are just superclasses and are known exceptions.
-            if event[0] not in ('ContainerRegistryArtifactEventData', 'ContainerRegistryEventData'):
+def extract(definitions):
+    if not definitions:
+        return
+    tups = []
+    for event in definitions:
+        print(event)
```
Comment (rakshith91) [inline sdk/eventgrid/azure-eventgrid/swagger/postprocess_eventnames.py:30]:
```suggestion
```

File: sdk/eventgrid/azure-eventgrid/swagger/postprocess_eventnames.py [hunk 2]
```diff
@@ -1,44 +1,41 @@
...
-            event_name = re.findall("Microsoft.[a-zA-Z]+.[a-zA-Z]+", event[1].__doc__)[0]
-        except:
-            # these two are just superclasses and are known exceptions.
-            if event[0] not in ('ContainerRegistryArtifactEventData', 'ContainerRegistryEventData'):
+def extract(definitions):
+    if not definitions:
+        return
+    tups = []
+    for event in definitions:
+        if event.endswith('Data') and event not in ('ContainerRegistryArtifactEventData', 'ContainerRegistryEventData'):
```
Comment (lmazuel) [inline sdk/eventgrid/azure-eventgrid/swagger/postprocess_eventnames.py:30]: I assume `('ContainerRegistryArtifactEventData', 'ContainerRegistryEventData')` is some kind of exception list, I would make it a more explicit configuration outside of the code for clarity

File: sdk/eventgrid/azure-eventgrid/swagger/postprocess_eventnames.py [hunk 3]
```diff
@@ -50,7 +47,10 @@ def generate_enum_content(tuples):
     print("# servicebus alias")
     print("ServiceBusDeadletterMessagesAvailableWithNoListenerEventName = 'Microsoft.ServiceBus.DeadletterMessagesAvailableWithNoListeners'")
```
Comment (lmazuel) [inline sdk/eventgrid/azure-eventgrid/swagger/postprocess_eventnames.py:48]: Why this line is all alone here?

File: sdk/eventgrid/azure-eventgrid/swagger/postprocess_eventnames.py [hunk 4]
```diff
@@ -1,44 +1,41 @@
...
-    'ResourceWriteCancelEventName': "Microsoft.Resources.ResourceWriteCancel",
-    'ResourceActionFailureEventName': "Microsoft.Resources.ResourceActionFailure",
-    'AcsChatMemberRemovedFromThreadWithUserEventName': "Microsoft.Communication.ChatMemberRemovedFromThreadWithUser",
-    'IoTHubDeviceConnectedEventName': "Microsoft.Devices.DeviceConnected",
-    'EventGridSubscriptionDeletedEventName': "Microsoft.EventGrid.SubscriptionDeletedEvent",
-    'AcsChatThreadParticipantRemovedEventName': "Microsoft.Communication.ChatThreadParticipantRemoved",
-    'ResourceActionCancelEventName': "Microsoft.Resources.ResourceActionCancel",
-    'IoTHubDeviceCreatedEventName': "Microsoft.Devices.DeviceCreated"
-}
+#===============================================deprecated===========================================
```
Comment (lmazuel) [inline sdk/eventgrid/azure-eventgrid/swagger/postprocess_eventnames.py:9]: If it's deprecated, remove it, Github is here to save the history

**Review by lmazuel — APPROVED** [review]

**Pull request merged**